src/ElasticAppSearchClient.php

```diff
@@ -46,33 +47,41 @@ public function __construct(ClientInterface $http_client, ConfigFactoryInterface
 
   }
 
-  public function getInstance($server){
+  /**
+   * {@inheritdoc}
+   */
+  public function getInstance($server) {
 
-    if(!empty($this->client)){
-      return $this->client;
+    if (!empty($this->client[$server->id()])) {
+      return $this->client[$server->id()];
     }
 
     $clientBuilder = ClientBuilder::create(
       $server->getHost(),
       $server->getSecret()
     );
 
-    $this->client = $clientBuilder->build();
+    $this->client[$server->id()] = $clientBuilder->build();
 
-    return $this->client;
+    return $this->client[$server->id()];
   }
 
-  public static function connect($server, $apikey, $engine = FALSE){
-    $clientBuilder = ClientBuilder::create($server, $apikey );
+  /**
+   * {@inheritdoc}
+   */
+  public static function connect($server, $apikey, $engine = FALSE) {
+    $clientBuilder = ClientBuilder::create($server, $apikey);
     $client = $clientBuilder->build();
-    if($engine){
+    if ($engine) {
       $engine = $client->getEngine($engine);
     }
     return $client;
```

Comment:
You are always returning `$client` even if `$engine` is set to true.

Reply:
fixed